### ISSUE TITLE
asformat mis-formats MXML Script tag & ignores contents

### ISSUE THREAD

**Issue by esidegallery** [BowlerHatLLC/vscode-as3mxml#611]
This is confirmed in a fresh installation of the latest VSCode with the ActionScript & MXML extension on Windows 10 x64. 

The opening Script tag is unindented and the ActionScript inside isn't formatted.

![ASFormat Indentation](https://user-images.githubusercontent.com/354051/179484760-077ef04f-c2fb-4eef-bc47-76e412a8ccf0.gif)

**Comment by esidegallery** [comment]
I've found that if I change the end-of-line sequence option in VS Code from CRLF to LF I can get full formatting to work. However, there seems to be a critical mass of code you can reach which causes the formatter to crash. I've attached the console output below as it's too long to paste:

[error.txt](https://github.com/BowlerHatLLC/vscode-as3mxml/files/9160217/error.txt)

It's hard to determine what exactly causes this threshold to be crossed, as it seems to differ depending on the content of the Script tag. Take the following file (I had to add a .txt extension to get it to upload):

[TileList.mxml.txt](https://github.com/BowlerHatLLC/vscode-as3mxml/files/9160694/TileList.mxml.txt)

Ensuring EOL is set to LF, I can get this file to crash the formatter by duplicating lines in the ListCollection block and then get it to work again by removing lines.

Empty lines don't affect this, but whitespace characters do seem to have an effect.

**Comment by joshtynjala** [comment]
Thanks. I may not have tested much with CRLF because I tend to avoid it, even on Windows. I'll check that out. I'll also investigate the crash, which seems to be from a regular expression.

**Comment by esidegallery** [comment]
I admit I never thought to change it from the default 'auto' setting in VS Code, but have done now!

**Comment by joshtynjala** [comment]
apache/royale-compiler@d76804aae02115dbb76c09eb3b39ab183281da5f

**Comment by esidegallery** [comment]
Sorry to be a pest, but just wondering if this looks fixable (either the CRLF or LF - whichever is easier!) It would be such a massive enhancement to my workflow if so.

**Comment by esidegallery** [comment]
Awesome! 🎉

**Comment by esidegallery** [comment]
Josh, it appears that MXML formatting is broken in a similar way for both CRLF and LF style line endings. If the Script contents is too long, formatting breaks, but removing lines will eventually allow it to work. The strange thing is that once it works, I can re-add lines and it will still work - up to a point at least. CRLF seems to be a little more sensitive than LF. This is a Windows PC btw. I'm using the file I posted above for the example below:

![Animation](https://github.com/BowlerHatLLC/vscode-as3mxml/assets/354051/f1979be5-81c7-48f4-ac70-e3b26816d617)

Shall I create a new issue for this?

**Comment by joshtynjala** [comment]
Yeah, create a new issue. Be sure to click the "Go to output" button and copy the exception that is reported.
